// your-app/src/routes/enrollCourse.route.js
const express = require('express');
const router = express.Router();
const enrollStudents = require('../models/enrollStudent');
const courses = require('../models/Courses');
const Students = require('../models/Students');
router.get("/getEnrolledCourses", (req, res, next) => {
  enrollStudents.find()
      .populate('books',['_id', 'book_id', 'book_name', 'author_name', 'quantity'])
      .populate('users',['id', 'name', 'email', 'phoneNo'])
      .exec()
      .then(docs => {
        res.status(200).json({
          enrollStudents: docs.map(doc => {
            return {
                _id: doc._id,
              courses: doc.courses,
              Students: doc.Students,
              enrollDate: doc.enrollDate.toDateString(),
              graduationDate: doc.graduationDate.toDateString()
            };
          })
        });
      })
      .catch(err => {
        res.status(500).json({
          error: err
        });
      });
  });

  router.post("/enrollStudent", (req, res, next) => {
    courses.find(req.body)
      .then(courses => {
        if (!courses) {
          return res.status(404).json({
            message: "Course Does Not Exist"
          });
        }
        const enrollStudent = new enrollStudents({
          enrollDate: req.body.enrollDate,
          graduationDate: req.body.graduationDate,
          courses: req.body,
          Students: req.body.id
        });
        return enrollStudent.save();
      })
      .then(result => {
        res.status(201).json({
          message: "Student Enrolled",
          enrolledStudent: {
            Students: result.Students,
            courses: result.courses,
            enrollDate: result.enrollDate,
            graduationDate: result.graduationDate
          }
        });
      })
      .catch(err => {
        console.log(err);
        res.status(500).json({
          error: err
        });
      });
  });
  

module.exports = router;
