// your-app/src/actions/authentication.js
import axios from 'axios';
import { GET_ERRORS, SET_CURRENT_USER, GET_BOOKCAT, GET_COURSES, DELETE_COURSE, 
    DELETE_BOOKCAT, FETCH_BOOKCAT, GET_ENROLLED_COURSES, GET_STUDENT} from './types';
import setAuthToken from '../setAuthToken';
import jwt_decode from 'jwt-decode';


export const registerUser = (user, history) => dispatch => { 
 console.log(user)

    axios.post('https://data.mongodb-api.com/app/data-nhrmr/endpoint/register',  user)
            .then(res => history.push('/login'))
            .catch(err => {
                dispatch({
                    type: GET_ERRORS,
                    payload: err.response.data
                });
            });
}

export const loginUser = (user) => dispatch => { 
    axios.post('https://data.mongodb-api.com/app/data-nhrmr/endpoint/login', user ) 
    
            .then(res => {
                const { token } = res.data;
                localStorage.setItem('jwtToken', token);
                setAuthToken(token);
                const decoded = jwt_decode(token);
                dispatch(setCurrentUser(decoded));
                alert("Successfully Logged In");
            })
            .catch(err => {
                dispatch({
                    type: GET_ERRORS,
                    payload: err.response.data
                });
            });
}

export const setCurrentUser = decoded => {
    return {
        type: SET_CURRENT_USER,
        payload: decoded
    }
}

export const logoutUser = (history) => dispatch => {
    localStorage.removeItem('jwtToken');
    localStorage.removeItem('roles');
    setAuthToken(false);
    dispatch(setCurrentUser({}));
    //history.push('/login');
}

// Get student User Details
export const getStudentUser = () => dispatch => {
    axios.get('/api/ClassManagement/getStudentUser')
        .then(res => {
            dispatch(fetchStudent(res.data));
        })
        .catch(err => {
            dispatch({
                type: GET_ERRORS,
                payload: err.response.data
            });
        });    
}

export const fetchStudent = payload => {
    return {
      type: GET_STUDENT,
      payload
    }
};

// Insert Books
export const newcourses = (courses,history) => dispatch => {    
    axios.post('/api/ClassManagement/addbooks', courses)
        .then(res => history.push('/'))
        .catch(err => {
            dispatch({
                type: GET_ERRORS,
                payload: err.response.data
            });
        });
}

// Insert Book category
export const bookcategory = (bookcats,history) => dispatch => {
axios.post('/api/ClassManagement/bookcat', bookcats)
        .then(res => {
            history.push('/category_manage');
        })
        .catch(err => {
            dispatch({
                type: GET_ERRORS,
                payload: err.response.data
            });
        });
}

// Fetch Book Category
export const getBookCat = () => dispatch => {
    axios.get('/api/ClassManagement/getbookcat')
        .then(res => {
            const schemas = [];
            let getcat = res.data;
            for(let i in getcat){
              schemas.push(getcat[i].book_cat);
            }
            dispatch(getBooksPosts(schemas));
        })
        .catch(err => {
            dispatch({
                type: GET_ERRORS,
                payload: err.response.data
            });
        });
}

export const getBooksPosts = payload => {
    return {
      type: GET_BOOKCAT,
      payload
    }
  };

//   Fetch Books
export const getCourses = () => dispatch => {
    axios.get('/api/ClassManagement/getbook')
        .then(res => {
            dispatch(fetchCourses(res.data));
        })
        .catch(err => {
            dispatch({
                type: GET_ERRORS,
                payload: err.response.data
            });
        });    
}

export const fetchCourses = payload => {
    return {
      type: GET_COURSES,
      payload
    }
  };

//   Delete Book
export const removeCourse = (_id) => dispatch => {
    axios.delete(`/api/ClassManagement/deleteBook/${_id}`) 
        .then(res => {
            dispatch(deleteBooksPost(res.data));

        })
        .catch(err => {
            dispatch({
                type: GET_ERRORS,
                payload: err.response.data
            });
        })
}

export const deleteBooksPost = _id =>{
    return {
        type: DELETE_COURSE,
        payload: {
            _id
        }
      }
}

// Delete Book Category
export const deleteBookCat = (_id,history) => dispatch => {
    axios.delete(`/api/ClassManagement/deleteBookCat/${_id}`) 
        .then(res => {
            dispatch(deleteBookCatPost(res.data));
            history.push('/');
        })
        .catch(err => {
            dispatch({
                type: GET_ERRORS,
                payload: err.response.data
            });
        })
}

export const deleteBookCatPost = _id =>{
    return {
        type: DELETE_BOOKCAT,
        payload: {
            _id
        }
      }
}

// Fetch Book Category for View Book Category
export const fetchBookCat = () => dispatch => {
    axios.get('/api/ClassManagement/getbookcat')
        .then(res => {
            dispatch(getBookCatPosts(res.data));
        })
        .catch(err => {
            dispatch({
                type: GET_ERRORS,
                payload: err.response.data
            });
        });
}

export const getBookCatPosts = payload => {
    return {
      type: FETCH_BOOKCAT,
      payload
    }
  };

// Issue Book- Student
export const enrollStudent = (courses,history) => dispatch => {    
    axios.post('/api/ClassManagement/issueBook', courses)
        .then(res => history.push('/student_course_manage'))
        .catch(err => {
            dispatch({
                type: GET_ERRORS,
                payload: err.response.data
            });
        });
}

//   Fetch Issued Books
export const getEnrolledCourses = () => dispatch => {
    axios.get('/api/ClassManagement/getissueBooks')
        .then(res => {
            dispatch(fetchEnrolledCourses(res.data));
        })
        .catch(err => {
            dispatch({
                type: GET_ERRORS,
                payload: err.response.data
            });
        });    
}

export const fetchEnrolledCourses = payload => {
    return {
      type: GET_ENROLLED_COURSES,
      payload
    }
  };
