// your-app/src/components/adminPages/ViewCourseDetails.js
import React, {Component} from 'react';
// import {Link} from 'react-router-dom';
import {getCourses} from '../../actions/authentication';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import axios from 'axios';

class ViewCourseDetails extends Component{
    constructor(props){
        super(props);
        this.state={
            course_id: "",
            course_name: "",
            teacher_name: "",
            book_cat: "",
            course_size: "",
            status: ""
        }
    }

    componentDidMount(){

        axios.get('/api/library/viewBookDetail/'+this.props.match.params.id)
            .then(res => {
                this.setState({ 
                  course_id: res.data.course_id, 
                  course_name: res.data.course_name,
                  teacher_name: res.data.teacher_name,
                  book_cat: res.data.book_cat,
                  course_size: res.data.course_size,
                  status: res.data.status,
                  editedBy: res.data.editedBy.name
                });
            })
            .catch((err) =>{
                console.log(err);
            })
    }

    render() {
        
        return(
            <div className="container">
                <div className="row">
                    <div className="col-md-12">
                        <div className="card">
                            <div className="card-header bg-success text-white"><strong>View Course Details</strong></div>
                            <div className="card-body">
                                <table className="table table-striped">
                                    <tbody>
                                        <tr>
                                            <td>Course ID</td>
                                            <td>{this.state.course_id}</td>
                                        </tr>
                                        <tr>
                                            <td>Course NAME</td>
                                            <td>{this.state.course_name}</td>
                                        </tr>
                                        <tr>
                                            <td>Course CATEGORY</td>
                                            <td>{this.state.book_cat}</td>
                                        </tr>
                                        <tr>
                                            <td>Teacher NAME</td>
                                            <td>{this.state.teacher_name}</td>
                                        </tr>
                                        <tr>
                                            <td>Course Size</td>
                                            <td>{this.state.course_size}</td>
                                        </tr>
                                        <tr>
                                            <td>EDITED BY</td>
                                            <td>{this.state.editedBy}</td>
                                        </tr>
                                        <tr>
                                            <td>STATUS</td>
                                            <td>{this.state.status.toString()}</td>
                                        </tr>
                                    </tbody>
                                </table>
                            </div>
                        </div>
                    </div>
                </div>
            </div>

        )
    }
}

const mapStateToProps = (state,props)=> ({
    auth: state.auth,
    errors: state.errors,
    courses: state.courses
});


export default connect(
    mapStateToProps,
    {getCourses}
  )(withRouter(ViewCourseDetails));
