// your-app/src/components/adminPages/ViewCourse.js
import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import {getCourses, removeCourse} from '../../actions/authentication';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import axios from 'axios';

class ViewCourse extends Component{

    constructor(props){
        super(props);
        this.state ={
            search: '',
            status: true
        }
    }

    updateSearch(event){

        this.setState({
            search: event.target.value.toLowerCase().substr(0,20)
        });
    }

    deleteCourse(event,_id,course_name) {
        event.preventDefault();
        if (
            window.confirm(
                `Do you want to delete  ${course_name} course?`,
            )
        ){   
            this.props.removeCourse(_id);
            this.props.getCourses();
       } 
    }

    componentDidMount(){
        this.props.getCourses();

    }

    activeCourse(event,_id,status){
        event.preventDefault();
        
        const activeCourse = {
            status: !status
        };

        axios.put('/api/library/activeBook/'+ _id, activeCourse)
            .then(res => {
                    this.props.getCourses();
                });
        
        
    }

    render() {
        const viewCourse = this.props.courses.viewCourse.filter(
            (el) =>
            {
                // return Object.keys(el).some(key =>
                //     el[key].toString().toLowerCase().includes(this.state.search.toLowerCase())
                //   );
                return Object.keys(el).some(key => el[key].toString().toLowerCase().search(this.state.search.toLowerCase()) !== -1);
            }
        );


        return(

                <div>
                    
                    <input type="text"
                    className="float-right searchBox"
                    placeholder="Search for..."
                    value={this.state.search}
                    onChange = {this.updateSearch.bind(this)}
                    />
                        
                    <table className="table table-hover table-bordered view-course">
                        <thead>
                            <tr>
                                <th>S.No.</th>
                                {/* <th>Book Id</th> */}
                                <th>Course Name</th>
                                <th>Class Size</th>
                                <th>Action</th>
                            </tr> 
                        </thead>
                        <tbody>
                        
                            {viewCourse.map((el) => {
                            return  <tr className="firstrow" key={el._id}>
                                        <td className="sno"></td>
                                        {/* <td>{el.book_id}</td> */}
                                        <td>{el.course_name.charAt(0).toUpperCase() + el.course_name.substring(1)}</td>
                                        <td>{el.book_cat.charAt(0).toUpperCase() + el.book_cat.substring(1)}</td>
                                        <td>{el.quantity}</td>
                                        <td><Link   to={"/editCourse/" + el._id} 
                                                    className="far fa-edit editCourse"
                                                    title="Edit Course"
                                            >
                                            </Link>
                                            <Link to='/' 
                                                onClick={ (event) => this.removeCourse(event,el._id, el.course_name) } 
                                                className="far fa-trash-alt deleteCourse"
                                                title="Delete Course"
                                            >
                                            </Link>
                                            <Link to={"/viewCourseDetails/" + el._id} 
                                                className="fas fa-address-book detailCourse"
                                                title="Course Details"
                                            >
                                            </Link>
                                            <button  
                                                onClick={ (event) => this.activeCourse(event,el._id,el.status) } 
                                                className={el.status === true ? "fas fa-eye activeElement": "fas fa-eye-slash activeElement"}
                                                title={el.status === true ? "Active" : "Deactive"}
                                            >
                                            </button>
                                        </td>
                                    </tr>
                            })}
                                        
                        </tbody>
                    </table>
                </div>


        )
    }
}

const mapStateToProps = state => ({
    auth: state.auth,
    errors: state.errors,
    courses: state.courses
});


export default connect(
    mapStateToProps,
    {getCourses,removeCourse}
  )(withRouter(ViewCourse));
