// your-app/src/components/adminPages/ViewStudentEnrolls.js
import React, { Component } from 'react';
import {getEnrolledCourses} from '../../actions/authentication';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

class ViewStudentEnrolls extends Component {

    componentDidMount(){
        this.props.getEnrolledCourses();
    }

    render(){

        this.courses = this.props['courses']['EnrolledCourses']['EnrollCourse']

        if( this.props['courses'] && this.props['courses']['EnrolledCourses'] && this.props['courses']['EnrolledCourses']['EnrollCourse'] ){
            var EnrollCourseStudent = [];
            this.courses.forEach(obj1 => {
                this.obj = obj1;
        
                if(this.props.match.params.id === this.obj.users._id ){
                    EnrollCourseStudent.push( this.obj);
                }
            });
        }
        
        return(

            <div>
                
                <table className="table table-hover table-bordered view-course">
                        <thead>
                            <tr>
                                <th>S.No.</th>
                                <th>Course Id</th>
                                <th>Course Name</th>
                                <th>Teacher Name</th>
                                <th>Enrollment Date</th>
                                <th>Graduation Date</th>
                            </tr> 
                        </thead>
                        <tbody>
                            
                            {EnrollCourseStudent && EnrollCourseStudent.map((el) => {
                            return  <tr className="firstrow" key={el.courses._id}>
                                        <td className="sno"></td>
                                        <td>{el.courses.course_id}</td>
                                        <td>{el.courses.course_name.charAt(0).toUpperCase() + el.courses.course_name.substring(1)}</td>
                                        <td>{el.courses.teacher_name.charAt(0).toUpperCase() + el.courses.teacher_name.substring(1)}</td>
                                        <td>{el.enrollmentDate}</td>
                                        <td>{el.graduationDate}</td>
                                    </tr>
                            })}
                                        
                        </tbody>
                    </table>
            </div>
        )
    }
}


const mapStateToProps = state => ({
    auth: state.auth,
    errors: state.errors,
    courses: state.courses
});

export default connect(
    mapStateToProps,
    {getEnrolledCourses}
  )(withRouter(ViewStudentEnrolls));
