// your-app/src/components/studentPages/DbStudent.js
import React, {Component} from 'react';
import {getEnrolledCourses} from '../../actions/authentication';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

class DbStudent extends Component{

    componentDidMount(){
        this.props.getEnrolledCourses();
    }

    render(){
        this.courses = this.props['courses']['enrolledCourses']['enrollCourse']

        if( this.props['courses'] && this.props['courses']['enrolledCourses'] && this.props['courses']['enrolledCourses']['enrollCourse'] ){
            var enrolledCoursesStudent = [];
            this.books.forEach(obj1 => {
                this.obj = obj1;
                if(this.obj.users._id === this.props.auth.user.id ){
                    enrolledCoursesStudent.push(this.obj.courses)
                }
            });
       
            // console.log('test', issuedBookuser)
        }
        
        return(
            
            <div className="dashboard">
                <div className="row">
                <table className="table table-hover table-bordered view-course">
                        <thead>
                            <tr>
                                <th>S.No.</th>
                                <th>Course Name</th>
                                <th>Teacher Name</th>
                                <th>Course Size</th>
                            </tr> 
                        </thead>
                        <tbody>
                        
                            {enrolledCoursesStudent && enrolledCoursesStudent.map((el) => {
                            return  <tr className="firstrow" key={el._id}>
                                        <td className="sno"></td>
                                        <td>{el.course_id}</td>
                                        <td>{el.course_name.charAt(0).toUpperCase() + el.course_name.substring(1)}</td>
                                        <td>{el.teacher_name.charAt(0).toUpperCase() + el.teacher_name.substring(1)}</td>
                                        <td>{el.course_size}</td>
                                    </tr>
                            })}
                                        
                        </tbody>
                    </table>
                </div>
            </div>
        )
    }
}

const mapStateToProps = state => ({
    auth: state.auth,
    errors: state.errors,
    courses: state.courses
});


export default connect(
    mapStateToProps,
    {getEnrolledCourses: getEnrolledCourses}
  )(withRouter(DbStudent));
